src/container/api.jl

```diff
@@ -18,13 +18,23 @@ abstract type ContainerInterface end
 
 """
 Default AgentAddress base type, where the agent identifier is based on the container created agent id (aid).
+Used with the TCP protocol.
 """
 @with_kw struct AgentAddress
     aid::Union{String,Nothing}
     address::Any = nothing
     tracking_id::Union{String,Nothing} = nothing
 end
 
+"""
```

Comment:
Maybe marker type for all address types? Then we do not need to repeat the send_message in agent core

Reply:
I agree to add a super type for addresses in the container API and change the send_message there accordingly.

This does not alleviate the need to change send_message for each subtype in the core, however. At least not without further rewriting and encapsulating the address-specific logic somewhere.

Sending a message to an MQTT broker is never locally forwarded and having that branch of the logic in there on the assumption that this
`if isnothing(receiver_addr) || receiver_addr == id(container.protocol)`
NEVER evaluates to true for some reason feels like it could lead to some hard to debug problems from the MQTT container side.